exonum-java-binding/qa-service/src/main/java/com/exonum/binding/qaservice/transactions/IncrementCounterTx.java

```diff
@@ -62,14 +64,16 @@ public IncrementCounterTx(long seed, HashCode counterId) {
   }
 
   @Override
-  public void execute(TransactionContext context) {
+  public void execute(TransactionContext context) throws TransactionExecutionException {
     QaSchema schema = new QaSchema(context.getFork());
     ProofMapIndexProxy<HashCode, Long> counters = schema.counters();
+
     // Increment the counter if there is such.
-    if (counters.containsKey(counterId)) {
-      long newValue = counters.get(counterId) + 1;
-      counters.put(counterId, newValue);
+    if (!counters.containsKey(counterId)) {
+      throw new TransactionExecutionException(UNKNOWN_COUNTER.code);
     }
+    long newValue = counters.get(counterId) + 1;
+    counters.put(counterId, newValue);
```

Comment:
is it time to add `computeIfPresent`? 🤔

Reply:
We do have a ticket for that, but I would put off any extensions till MerkleDb: https://jira.bf.local/browse/ECR-586?jql=text%20~%20%22contains%22%20and%20reporter%20%3D%20currentUser()%20